rust/src/lib.rs

```diff
@@ -3277,6 +3180,324 @@ pub unsafe extern "C" fn zcashlc_create_proposed_transactions(
+    NotInMainChain,
+    /// The requested transaction ID corresponds to a transaction that has been included in the
+    /// block at the provided height.
+    Mined(u32),
+}
+
+/// Sets the transaction status to the provided value.
+///
+/// # Safety
+///
+/// - `db_data` must be non-null and valid for reads for `db_data_len` bytes, and it must
+///   have an alignment of `1`. Its contents must be a string representing a valid system
+///   path in the operating system's preferred representation.
+/// - The memory referenced by `db_data` must not be mutated for the duration of the
+///   function call.
+/// - The total size `db_data_len` must be no larger than `isize::MAX`. See the safety
+///   documentation of pointer::offset.
+/// - `txid_bytes` must be non-null and valid for reads for `db_data_len` bytes, and it must have
+///   an alignment of `1`.
+/// - The memory referenced by `txid_bytes_len` must not be mutated for the duration of the
+///   function call.
+/// - The total size `txid_bytes_len` must be no larger than `isize::MAX`. See the safety
+///   documentation of pointer::offset.
+#[no_mangle]
+pub unsafe extern "C" fn zcashlc_set_transaction_status(
+    db_data: *const u8,
+    db_data_len: usize,
+    network_id: u32,
+    txid_bytes: *const u8,
+    txid_bytes_len: usize,
+    status: FfiTransactionStatus,
+) {
+    let res = catch_panic(|| {
+        let network = parse_network(network_id)?;
+        let mut db_data = unsafe { wallet_db(db_data, db_data_len, network)? };
+
+        let txid_bytes = unsafe { slice::from_raw_parts(txid_bytes, txid_bytes_len) };
+        let txid = TxId::read(&txid_bytes[..])?;
+
+        let status = match status {
+            FfiTransactionStatus::TxidNotRecognized => TransactionStatus::TxidNotRecognized,
+            FfiTransactionStatus::NotInMainChain => TransactionStatus::NotInMainChain,
+            FfiTransactionStatus::Mined(h) => TransactionStatus::Mined(BlockHeight::from(h)),
+        };
+
+        db_data
+            .set_transaction_status(txid, status)
+            .map_err(|e| anyhow!("Error setting transaction status for txid {}: {}", txid, e))
+    });
+
+    unwrap_exc_or(res, ())
+}
+
+/// A request for transaction data enhancement, spentness check, or discovery
+/// of spends from a given transparent address within a specific block range.
+#[repr(C, u8)]
+pub enum FfiTransactionDataRequest {
+    /// Information about the chain's view of a transaction is requested.
+    ///
+    /// The caller evaluating this request on behalf of the wallet backend should respond to this
+    /// request by determining the status of the specified transaction with respect to the main
+    /// chain; if using `lightwalletd` for access to chain data, this may be obtained by
+    /// interpreting the results of the [`GetTransaction`] RPC method. It should then call
+    /// [`WalletWrite::set_transaction_status`] to provide the resulting transaction status
```

Comment:
Should this not say to call the corresponding FFI method? (And similarly below.)